docs/getting-started/rulesets.md

```diff
@@ -0,0 +1,141 @@
+# Spectral Rulesets
+
+Rulesets are collections of rules, in a YAML or JSON file. These rules are taking parameters, and calling functions on certain parts of another YAML or JSON object being linted. 
+
+### Adding a rule
+
+Add your own rules under the `rules` property in your `.spectral.yml` ruleset file.
+
+```yaml
+rules:
+  my-rule-name:
+    description: Tags must have a description.
+    given: $.tags[*]
+    severity: error
+    then:
+      field: description
+      function: truthy
+```
+
+The example above adds a single rule that checks that tags objects have a description property defined.
+
+Running `spectral lint` on the following object with the ruleset above will result in an error being reported, since the tag does not have a description:
+
+```json
+{
+  "tags": [{
+    "name": "animals"
+  }]
+}
+```
+
+While running it with this object, it will succeed:
+
+```json
+{
+  "tags": [{
+    "name": "animals",
+    "description": "come in all shapes and sizes"
+  }]
+}
+```
+
+### Severity
+
+The `severity` keyword is optional and can be `error`, `warn`, `info`, or `hint`.
+
+### Then
+
+The Then part of the rules explains what to do with the `given` JSON Path, and involves two required keywords:
+
+```yaml
+  then:
+    field: description
+    function: truthy
+```
+
+The `field` keyword is the name for a property in the object selected. It could also be `@key` if the given path is an object.
+
+```yaml
+given: '$.responses'
+then:
+  field: '@key'
+  function: pattern
+  functionOptions:
+    match: '^[0-9]+$' 
+```
+
+The above pattern based rule would error on `456avbas` as it is not numeric.
+
+```yaml
+responses:
+  123:
+    foo: bar
+  456avbas:
+    foo: bar
+```
+
+### Extending Rules
+
+Rulesets can extend other rulesets. For example, Spectral comes with two built in rulesets - one for OpenAPI v2 (`spectral:oas2`), and one for OpenAPI v3 (`spectral:oas3`). 
+
+Use the `extends` property in your ruleset file to build upon or customize other rulesets.
+
+```yaml
+extends: spectral:oas2
+rules:
+  my-rule-name:
+    description: Tags must have a description.
+    given: $.tags[*]
+    then:
+      field: description
+      function: truthy
+```
+
+The example above will apply the core rules from the built in OpenAPI v2 ruleset AND apply the custom `my-rule-name` rule. 
+
+Extends can be a single string or an array of strings, and can contain either local file paths or URLs. In the future we will support loading from NPM modules. 
```

Comment:
Have you considered adding an example showcasing loading ruleset from a URL?

Reply:
Nice, done!